File: CS/HowToUseTreeListEditor.Module/Project.cs
using System;

using DevExpress.Xpo;

using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;
using DevExpress.Persistent.Base.General;
using System.ComponentModel;

namespace HowToUseTreeListEditor.Module {
    public class Project : Category {
        private ProjectGroup projectGroup;
        protected override ITreeNode Parent {
            get {
                return projectGroup;
            }
        }
        protected override IBindingList Children {
            get {
                return ProjectAreas;
            }
        }
        public Project(Session session) : base(session) { }
        public Project(Session session, string name)
            : base(session) {
            this.Name = name;
        }
        [Association("ProjectGroup-Projects")]
        public ProjectGroup ProjectGroup {
            get {
                return projectGroup;
            }
            set {
                SetPropertyValue("ProjectGroup", ref projectGroup, value);
            }
        }
        [Association("Project-ProjectAreas"), Aggregated]
        public XPCollection<ProjectArea> ProjectAreas {
            get {
                return GetCollection<ProjectArea>("ProjectAreas");
            }
        }
    }


}

File: VB/HowToUseTreeListEditor.Module/Project.vb
Imports DevExpress.Xpo
Imports DevExpress.Persistent.Base.General
Imports System.ComponentModel

Namespace HowToUseTreeListEditor.Module

    Public Class Project
        Inherits Category

        Private projectGroupField As ProjectGroup

        Protected Overrides ReadOnly Property ParentProp As ITreeNode
            Get
                Return projectGroupField
            End Get
        End Property

        Protected Overrides ReadOnly Property ChildrenProp As IBindingList
            Get
                Return ProjectAreas
            End Get
        End Property

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub

        Public Sub New(ByVal session As Session, ByVal name As String)
            MyBase.New(session)
            NameProp = name
        End Sub

        <Association("ProjectGroup-Projects")>
        Public Property ProjectGroup As ProjectGroup
            Get
                Return projectGroupField
            End Get

            Set(ByVal value As ProjectGroup)
                SetPropertyValue("ProjectGroup", projectGroupField, value)
            End Set
        End Property

        <Association("Project-ProjectAreas"), Aggregated>
        Public ReadOnly Property ProjectAreas As XPCollection(Of ProjectArea)
            Get
                Return GetCollection(Of ProjectArea)("ProjectAreas")
            End Get
        End Property
    End Class
End Namespace
